File: news-api/src/main/java/com/newsarcticlesapp/newsapi/listener/model/Article.java
package com.newsarcticlesapp.newsapi.listener.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class Article {
    @JsonProperty("title")
    private String title;

    @JsonProperty("article_type")
    private String articleType;

    @JsonProperty("text")
    private String text;

    @JsonProperty("keywords")
    private List<String> keywords;

    @JsonProperty("datetime")
    private LocalDateTime datetime;

    @JsonProperty("source")
    private String source;

    @JsonProperty("url")
    private String url;

    @JsonProperty("number_of_views")
    private int numberOfViews;

    @JsonProperty("picture_link")
    private String pictureLink;

    // Getters and setters
}

File: aggregation-api/app/model/article.py
class Article:

    def __init__(self, title, article_type, text, keywords, datetime, source, url, number_of_views, picture_link):
        self.title = title
        self.article_type = article_type
        self.text = text
        self.keywords = keywords
        self.create_at = datetime
        self.source = source
        self.url = url
        self.number_of_views = number_of_views
        self.picture_link = picture_link

    def to_dict(self):
        return {
            "title": self.title,
            "article_type": self.article_type,
            "text": self.text,
            "keywords": self.keywords,
            "datetime": self.create_at.isoformat() if hasattr(self.create_at, "isoformat") else str(self.create_at),
            "source": self.source,
            "url": self.url,
            "number_of_views": self.number_of_views,
            "picture_link": self.picture_link,
        }
